tokio/src/task/join_set.rs

```diff
@@ -374,6 +374,26 @@ impl<T: 'static> JoinSet<T> {
         while self.join_next().await.is_some() {}
     }
 
+    /// Awaits the completion of all tasks on this `JoinSet`, returning a Vector of their results.
+    ///
+    /// If any task on this `JoinSet` errors or panics, this will panic and cancel all
```

Comment:
I think "cancel" might be more appropriate than "error"?

```suggestion
    /// If any task on this `JoinSet` gets cancelled or panics, this will panic and cancel all
```